index.js

```diff
@@ -1,10 +1,187 @@
+            console.log(`resolved persons ${JSON.stringify(result)}`)
+            console.log(Person.find())
+            res.json(result.map(person => person.toJSON()))
+            // console.log(person)
+        })
+        // .then(result => {
+        //     res.status(204).end()
+        // })
+        // eslint-disable-next-line no-unused-vars
+        .catch((error) => {
+            res.status(204).end()
+        })
+})
+
+// darbojas
+app.get('/info', (req, res) => {
+    Person.find({}).then(result => {
+        res.send(`<p>Phonebook has info for ${result.length} people</p>
+    <p>${new Date}</p>`)
+    })
+})
+
+// darbojas
+app.get('/api/persons/:id', (req, res, next) => {
+    Person.findById(req.params.id)
+        .then(person => {
+            if (person) {
+                res.json(person)
+            } else {
+                res.status(404).end()
+            }
+        })
+        .catch(error => next(error))
+})
+
+// darbojas
+app.delete('/api/persons/:id', (req, res, next) => {
+    Person.findByIdAndRemove(req.params.id)
+        // eslint-disable-next-line no-unused-vars
+        .then(result => {
+            res.status(204).end()
+        })
+        .catch(error => next(error))
 })
 
-const PORT = 3001
-app.listen(PORT)
-console.log(`Server running on port ${PORT}`)
\ No newline at end of file
+// darbojas
+app.post('/api/persons', (req, res, next) => {
+    // const generateId = (max) =>
+    //     Math.floor(Math.random() * Math.floor(max))
+
+    const body = req.body
+
+    // if (!body.name) {
+    //     return res.status(400).json({
+    //         error: 'name missing'
+    //     })
+    // }
+
+    // if (!body.number) {
+    //     return res.status(400).json({
+    //         error: 'number missing'
+    //     })
+    // }
+    // if (persons.find(person => person.name === body.name)) {
+    //     return res.status(400).json({
+    //         error: 'name must be unique'
+    //     })
+    // }
+
+    const person = new Person({
+        name: body.name,
+        number: body.number,
+        // id: generateId(10000)
+    })
+
+    person.save().
+        then(savedPerson => {
+            res.json(savedPerson)
+        })
+        .catch(error => next(error))
+})
+
+app.put('/api/persons/:id', (req, res, next) => {
+    const body = req.body
+
+    const person = {
+        name: body.name,
+        number: body.number,
+    }
+
+    Person.findByIdAndUpdate(req.params.id, person, { new: true })
+        .then(updatedPerson => {
+            res.json(updatedPerson)
+        })
+        .catch(error => next(error))
+})
+
+// const unknownEndpoint = (req, res) => {
+//     res.status(404).send({ error: 'unknown endpoint' })
+// }
+
+// app.use(unknownEndpoint)
+
+
+const errorHandler = (error, req, res, next) => {
+    console.error(error.message)
+
+    // eslint-disable-next-line eqeqeq
+    if (error.name === 'CastError' && error.kind == 'ObjectId') {
```

Comment:
[Why did you even get the eqeqeq in the first place?](https://eslint.org/docs/rules/eqeqeq#require-and-eqeqeq:~:text=It%20is%20considered%20good%20practice%20to,their%20regular%20counterparts%20%3D%3D%20and%20!%3D.)